bootstrap/src/lib/cache/index.ts

```diff
@@ -76,7 +91,7 @@ export const withCache: Middleware<CacheOptions> = async (execute, options = def
   if (!options.enabled) return (data: AdapterRequest) => execute(data)
 
   const cache = await options.cacheBuilder(options.cacheOptions)
-
+  const rateLimit = makeRateLimit(options.rateLimit)
```

Comment:
This is now strongly coupled to our cache, it's inside our cache, but I don't really see a reason why?

We shouldn't have made any changes to this file and instead implement the RL as a completely separate component. It should just be a middleware, something more like this:

```javascript
// read part
const reqPerMin  = rateLimit.getTroughput(input, RateLimit.MINUTE)
const maxAge = _computeMaxAge(reqPerMin)
// this is withCache(execute)
const result = await execute({ ...input, maxAge })
// write part
if (!_isWarmer(input)) rateLimit.observe(input)
```

Reply:
Well before made sense, as the cache was storing everything related to Rate Limit Groups. I don't see the point of keeping it there now either.